lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -400,7 +400,7 @@ class AwsCompileFunctions {
         // Try convert reservedConcurrency to integer
         const reservedConcurrency = Number(functionObject.reservedConcurrency);
 
-        if (_.isInteger(reservedConcurrency)) {
+        if (typeof reservedConcurrency === 'number' && !isNaN(reservedConcurrency)) {
```

Comment:
Just use `isNaN` check (note that value is guraanteed there to be type of `number` as line above coercion to number takes place)